python/strings_udf/cpp/include/cudf/strings/udf/search.cuh

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/strings/string_view.cuh>
+
+namespace cudf {
+namespace strings {
+namespace udf {
+
+/**
+ * @brief Returns the number of times that the target string appears
+ * in the source string.
+ *
+ * @param source Source string to search
+ * @param target String to match within source
+ * @param start First character position within source to start the search
+ * @param end Last character position (exclusive) within source to search
+ * @return Number of matches
+ */
+__device__ inline cudf::size_type count(string_view const source,
+                                        string_view const target,
+                                        cudf::size_type start = 0,
+                                        cudf::size_type end   = -1)
+{
+  auto const tgt_length = target.length();
+  auto const src_length = source.length();
+
+  start = start < 0 ? 0 : start;
+  end   = (end < 0 || end > src_length) ? src_length : end;
```

Comment:
@davidwendt we should document the behavior of this function for values outside the boundaries in the docstring. I don't have any issue with this behavior, but it may not be what people are immediately expecting.